lib/remote.provider.js

```diff
@@ -1,7 +1,113 @@
 'use strict';
 
+const jstp = require('@metarhia/jstp');
+
 const { StorageProvider } = require('./provider');
+const { RemoteCursor } = require('./remote.cursor');
+
+const APP_NAME = 'GLOBAL_STORAGE';
+const INTERFACES = ['provider', 'cursor'];
+
+class RemoteProvider extends StorageProvider {
+  // Create RemoteProvider
+  // cursor - RemoteCursor instance
+  // apiPromise - jstp promise
+  constructor() {
+    super();
+    this.api = null;
+    this.connection = null;
+    this.cursors = [];
+  }
+
+  // Open StorageProvider
+  //   callback - function(err, RemoteProvider)
+  // eslint-disable-next-line no-unused-vars
+  open(options, callback) {
+    jstp.ws.connectAndInspect(
```

Comment:
Transport should be passed in options, it shouldn't be hardcoded as `ws`.
Also, I don't think we should use `connectAndInspect()` method generating `jstp.RemoteProvider`, this just adds additional overhead. You can use `connect()` and then `connection.callMethod()` to call remote methods.